File: src/util/benchmark/BenchmarkTimingsSummary.php
<?php

namespace pocketcloud\cloud\util\benchmark;

use pocketcloud\cloud\util\FormatUtils;

final readonly class BenchmarkTimingsSummary {

    public function __construct(
        private string $name,
        private int $count,
        private float $avg,
        private float $min,
        private float $max
    ) {}

    public function format(int $precision = 3): string {
        return sprintf(
            "Name: %s | Count: %s | Avg: %s | Min: %s | Max: %s",
            $this->name ?? "N/A",
            $this->count,
            FormatUtils::milliseconds($this->avg, $precision),
            FormatUtils::milliseconds($this->min, $precision),
            FormatUtils::milliseconds($this->max, $precision),
        );
    }

    public function getName(): string {
        return $this->name;
    }

    public function getCount(): int {
        return $this->count;
    }

    public function getAvg(): float {
        return $this->avg;
    }

    public function getMin(): float {
        return $this->min;
    }

    public function getMax(): float {
        return $this->max;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/util/benchmark/BenchmarkTimingsSummary.java
package de.pocketcloud.cloud.util.benchmark;

public record BenchmarkTimingsSummary(String name, long count, double avg, double min, double max, long lastTick) {

    public String format(int precision, boolean colorful) {
        String fmt = "%." + precision + "fms";
        if (colorful) {
            return String.format(
                    "Name: §b%s §8| §rCount: §b%d §8| §rLast: §btick %d §8| §rAvg: §b%s §8| §rMin: §b%s §8| §rMax: §b%s",
                    name,
                    count,
                    lastTick,
                    String.format(fmt, avg),
                    String.format(fmt, min),
                    String.format(fmt, max)
            );
        } else {
            return String.format(
                    "Name: %s | Count: %d | Last: tick %d | Avg: %s | Min: %s | Max: %s",
                    name,
                    count,
                    lastTick,
                    String.format(fmt, avg),
                    String.format(fmt, min),
                    String.format(fmt, max)
            );
        }
    }

    public String format(int precision) {
        return format(precision, false);
    }

    public String format() {
        return format(3, false);
    }
}